flake8_trio.py

```diff
@@ -55,19 +83,73 @@ def __init__(self) -> None:
         self.safe_yields: Set[ast.Yield] = set()
         self._yield_is_error = False
         self._context_manager = False
+        self._inside_finally: Optional[ast.Try] = None
+        self._scope: Optional[TrioScope] = None
 
     def visit_generic_with(self, node: Union[ast.With, ast.AsyncWith]):
         self.check_for_trio100(node)
 
-        outer = self._yield_is_error
-        if not self._context_manager and any(
-            is_trio_call(item, "open_nursery", *cancel_scope_names)
-            for item in (i.context_expr for i in node.items)
+        outer_yie = self._yield_is_error
+        outer_scope = self._scope
+
+        # Check for a `with trio.<scope_creater>`
+        trio_scope = next(
+            (
+                get_trio_scope(item, "open_nursery", *cancel_scope_names)
+                for item in (i.context_expr for i in node.items)
+            ),
+            None,
+        )
```

Comment:
The coverage issue is an uncovered _branch_ `96 -> exit`.  When I see this it's usually that you defined a lambda which was never called, but there's no lambda here!

I'm pretty sure it's that we never exhaust the iterable, and therefore never use the `None` fallback value - because `get_trio_scope` returns None if it's not a trio scope.  Incidentally I think this means we'll only consider the first item and therefore fail to detect the cancel scope in e.g. `with open(""), trio.fail_at(0):`.

At that point it's probably easier to write this as an explicit for-loop with a `break` 😅

Reply:
Ooh right, of course. Very nice that the coverage caught that issue. Thanks!
And yeah, added a case for that which previous version of the code did not handle.

Preposterous! Do you really think
```python
trio_scope = None
for item in (i.context_expr for i in node.items):
    trio_scope = get_trio_scope(item, "open_nursery", *cancel_scope_names)
    if trio_scope is not None:
        break
```
is better than
```python                                                                  
trio_scope = next(
   filter(
       None,
       (
           get_trio_scope(item, "open_nursery", *cancel_scope_names)
           for item in (i.context_expr for i in node.items)
       ),
   ),
   None,
)
```
??? :grin: